content/pedersen-polynomial-commitment/en/pedersen-polynomial-commitment.md

```diff
@@ -78,6 +78,34 @@ In a sense, the prover is evaluating the polynomial using the coefficients to th
 
 The proof of correct evaluation is that the prover is able to separate the blinding terms from evaluation of the polynomial -- even though the prover does not know the discrete logs of $yG$ and $\pi B$.
 
+``` python
+import random
+from functools import reduce
+from py_ecc.bn128 import G1, add, eq, field_modulus, multiply
+
+x, y = random.randint(1, field_modulus - 1), random.randint(1, field_modulus - 1)
+G, B = multiply(G1, x), multiply(G1, y)
+
+def commit(secret, salt):
+    return add(multiply(G, secret), multiply(B, salt))
+
+# Let's say polynomial be f(x) = x^2 + 2x + 3
+salts = [random.randint(1, field_modulus - 1) for _ in range(3)]
+u = random.randint(1, field_modulus - 1)
```

Comment:
technically, u is generated after the commitments in the algorithm. It might be helpful to put `u` below the commitments and add a comment

`# verifier receives the commitments and responds with u`